sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -1655,6 +1655,14 @@ object SQLConf {
       .checkValues(Dialect.values.map(_.toString))
       .createWithDefault(Dialect.SPARK.toString)
 
+  val DIALECT_SPARK_ANSI_ENABLED = buildConf("spark.sql.dialect.spark.ansi.enabled")
```

Comment:
Having the word `spark` twice seems overkill, isn't it?

Reply:
+1

Reply:
Thanks for the advice, done in d76d6a5

Reply:
I actually think `spark.sql.dialect.spark.ansi.enabled` makes sense because we can have multiple different directs that can be configured by `spark.sql.dialect` configuration.

For instance, if we happen to have another configuration specific to PostgreSQL dialect, we might have to have a configuration as `spark.sql.dialect.postgresql.blahblah`.

Reply:
It a bit look redundant for now if we'll only have `spark` but should be fine as `spark.sql.dialect.<value>` with an explicit plan.

Reply:
+1 with @HyukjinKwon 
`spark.sql.dialect.spark.ansi.enabled` is more clear, considering we have another option `spark.sql.dialect` which has two choices `spark` and `postgresql`

Reply:
Thanks all for the discussion, I change it back for clear semantics in next commit.